akka-typed/src/test/scala/akka/typed/PerformanceSpec.scala

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.typed
+
+import ScalaDSL._
+import Ops._
+import AskPattern._
+import StepWise._
+import scala.concurrent.duration._
+import scala.concurrent.Await
+import scala.concurrent.Future
+import com.typesafe.config.ConfigFactory
+
+class PerformanceSpec extends TypedSpec {
+
+  object `A static behavior` {
+
+    object `must be fast` {
+
+      case class Ping(x: Int, pong: ActorRef[Pong], report: ActorRef[Pong])
+      case class Pong(x: Int, ping: ActorRef[Ping], report: ActorRef[Pong])
+
+      def behavior(pairs: Int, pings: Int, count: Int, executor: String) =
+        StepWise[Pong] { (ctx, startWith) ⇒
+          startWith {
+            import ctx.executionContext
+
+            val pinger = Props(SelfAware[Ping](self ⇒ Static { msg ⇒
+              if (msg.x == 0) {
+                msg.report ! Pong(0, self, msg.report)
+              } else msg.pong ! Pong(msg.x - 1, self, msg.report)
+            })).withDispatcher(executor)
+
+            val ponger = Props(SelfAware[Pong](self ⇒ Static { msg ⇒
+              msg.ping ! Ping(msg.x, self, msg.report)
+            })).withDispatcher(executor)
+
+            val actors =
+              for (i ← 1 to pairs)
+                yield (ctx.spawn(pinger, s"pinger-$i"), ctx.spawn(ponger, s"ponger-$i"))
+
+            val start = Deadline.now
+
+            for {
+              (ping, pong) ← actors
+              _ ← 1 to pings
+            } ping ! Ping(count, pong, ctx.self)
+
+            start
+          }.expectMultipleMessages(15.seconds, pairs * pings) { (msgs, start) ⇒
+            val stop = Deadline.now
+
+            val rate = 2L * count * pairs * pings / (stop - start).toMillis
+            info(s"messaging rate was $rate/ms")
```

Comment:
Hm, consider JMH later? Or this is just for a rough validation (like I have with the interpreter)?